detection-rules/open_redirect_weblinkconnect.yml

```
@@ -0,0 +1,22 @@
+name: "Open redirect: weblinkconnect.com"
+description: "Message contains use of the weblinkconnect.com open redirect, but the sender is not weblinkconnect.com. This has been exploited in the wild."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          .href_url.domain.domain == 'api-internal.weblinkconnect.com'
+          and strings.icontains(.href_url.path, "/click")
+          and strings.icontains(.href_url.query_params, "url=")
+  )
+  and sender.email.domain.root_domain != 'weblinkconnect.com'
+
```

include high trust sender negation? 

```suggestion
  // negate highly trusted sender domains unless they fail DMARC authentication
  and (
    (
      sender.email.domain.root_domain in $high_trust_sender_root_domains
      and not headers.auth_summary.dmarc.pass
    )
    or sender.email.domain.root_domain not in $high_trust_sender_root_domains
  )
```